src/contract.rs

```diff
@@ -1659,9 +1640,7 @@ fn perform_transfer(
     BalancesStore::save(store, from, from_balance)?;
 
     let mut to_balance = BalancesStore::load(store, to);
-    to_balance = to_balance.checked_add(amount).ok_or_else(|| {
-        StdError::generic_err("This tx will literally make them too rich. Try transferring less")
-    })?;
+    safe_add(&mut to_balance, amount);
```

Comment:
maybe check here if the transfer amount was saturated, and if so update the total amount (although that may reveal that it was saturated, since two keys were accessed, so maybe just pass the maximum amount that would not make it saturate)